plugin/advancedsubscription/src/AdvancedSubscriptionPlugin.php

```diff
@@ -0,0 +1,872 @@
+<?php
+/* For licensing terms, see /license.txt */
+/**
+ * @TODO: Improve description
+ * This class is used to add an advanced subscription allowing the admin to
+ * create user queues requesting a subscribe to a session
+ * @package chamilo.plugin.advancedsubscription
+ */
+
+
+class AdvancedSubscriptionPlugin extends Plugin implements HookPluginInterface
+{
+    /**
+     * Constructor
+     */
+    function __construct()
+    {
+        $parameters = array(
+            'tool_enable' => 'boolean',
+            'yearly_cost_limit' => 'text',
+            'yearly_hours_limit' => 'text',
+            'yearly_cost_unit_converter' => 'text',
+            'courses_count_limit' => 'text',
+            'course_session_credit_year_start_date' => 'text',
+            'ws_url' => 'text',
+            'min_profile_percentage' => 'text',
+            'check_induction' => 'boolean',
+            'confirmation_message' => 'wysiwyg',
+            'secret_key' => 'text',
+        );
+
+        parent::__construct('1.0', 'Imanol Losada, Daniel Barreto', $parameters);
+    }
+
+    /**
+     * Instance the plugin
+     * @staticvar null $result
+     * @return AdvancedSubscriptionPlugin
+     */
+    static function create()
+    {
+        static $result = null;
+
+        return $result ? $result : $result = new self();
+    }
+
+    /**
+     * Install the plugin
+     * @return void
+     */
+    public function install()
+    {
+        $this->installDatabase();
+        $this->installHook();
+    }
+
+    /**
+     * Uninstall the plugin
+     * @return void
+     */
+    public function uninstall()
+    {
+        $this->uninstallHook();
+        $this->uninstallDatabase();
+    }
+
+    /**
+     * Create the database tables for the plugin
+     * @return void
+     */
+    private function installDatabase()
+    {
+        $pAdvSubQueueTable = Database::get_main_table(TABLE_ADV_SUB_QUEUE);
+
+        $sql = "CREATE TABLE IF NOT EXISTS $pAdvSubQueueTable (" .
+            "id int UNSIGNED NOT NULL AUTO_INCREMENT, " .
+            "session_id int UNSIGNED NOT NULL, " .
+            "user_id int UNSIGNED NOT NULL, " .
+            "status int UNSIGNED NOT NULL, " .
+            "last_message_id int UNSIGNED NOT NULL, " .
+            "created_at datetime NOT NULL, " .
+            "updated_at datetime NULL, " .
+            "PRIMARY KEY PK_advsub_queue (id), " .
+            "UNIQUE KEY UK_advsub_queue (user_id, session_id)); ";
+        Database::query($sql);
+    }
+
+    /**
+     * Drop the database tables for the plugin
+     * @return void
+     */
+    private function uninstallDatabase()
+    {
+        /* Drop plugin tables */
+        $pAdvSubQueueTable = Database::get_main_table(TABLE_ADV_SUB_QUEUE);
```

Comment:
Use complete words and don't add prefixes ($p)
$advancedSubscriptionQueueTable instead of $pAdvSubQueueTable  

The same with constants:
TABLE_ADVANCED_SUBSCRIPTION_QUEUE
instead of 
TABLE_ADV_SUB_QUEUE